### PR TITLE
make logout functional and more robust

### PR DESCRIPTION
**Pull request by pbugni**
Previously, the system would get in a stuck state when out of sync with keycloak login status - only cure was browser incognito mode.

Primary problem looks to have been the default session (if you remove all enhancements in this PR to include flask-session and use a non-default session type, it's easy to reproduce the problem by logging in, requesting /logout and then attempting to log in again).

### CODE REVIEW

File: patientsearch/config.py [hunk 1]
```diff
@@ -11,6 +11,7 @@ def decode_json_config(potential_json_string):
 
 SERVER_NAME = os.getenv("SERVER_NAME")
 SECRET_KEY = os.getenv("SECRET_KEY")
+SESSION_TYPE = os.getenv("SESSION_TYPE", 'filesystem')
```
Comment (pbugni) [inline patientsearch/config.py:14]: @ivan-c I suspect you won't love the choice of `filesystem` here - it is simply the lowest bar for a functional session.  The `null` default causes the same problem named in PR description.  Additional options require additional libs, but something like `redis` probably makes more sense.

File: patientsearch/api.py [hunk 2]
```diff
@@ -203,10 +204,19 @@ def external_search(resource_type, methods=["GET"]):
...
-    current_app.logger.info(
-        "logout on request",
-        extra={'tags': ['logout'], 'user_id': user_id})
-    terminate_session()
+    try:
+        user_id = oidc.user_getfield('email')
+        current_app.logger.info(
+            "logout on request",
+            extra={'tags': ['logout'], 'user_id': user_id})
+        terminate_session()
```
Comment (ivan-c) [inline patientsearch/api.py:212]: Unrelated, but could this POST to Keycloak be the reason we were regenerating a cookie at logout? I see `session.clear()` is called AFTER, so maybe not...

File: patientsearch/api.py [hunk 3]
```diff
@@ -203,10 +204,19 @@ def external_search(resource_type, methods=["GET"]):
...
+
+    except Exception as ex:
```
Comment (ivan-c) [inline patientsearch/api.py:214]: Lame they don't have a custom exception to catch here :/. Looks like you could [check `flask.g`](https://github.com/puiterwijk/flask-oidc/blob/master/flask_oidc/__init__.py#L241-L242) if you want something slightly cleaner

File: patientsearch/config.py (hunk 1 above)
Comment (ivan-c) [reply patientsearch/config.py:14]: No big deal, we can always add redis later if needed. What's another service :grin:

**Review by ivan-c — APPROVED** [review]
Ecstatic to have a stale session hole plugged! Merge when ready

File: patientsearch/api.py (hunk 2 above)
Comment (pbugni) [reply patientsearch/api.py:212]: `session.clear()` did not fix the symptom and does not appear to be required (once the session type is in place).  That was an early attempt, which i left in tact as it feels like a good idea.

File: patientsearch/api.py (hunk 3 above)
Comment (pbugni) [reply patientsearch/api.py:214]: flask.g is hardly cleaner :)  i concur that it's lame.  this should never happen regardless, only found when debugging the bogus session, but leaving in tact as it does make it more robust.

**Pull request merged**